tools/ports/freetype.py

```diff
@@ -1,602 +1,34 @@
-              'src/base/ftbdf.c',
-              'src/base/ftbitmap.c',
-              'src/base/ftcalc.c',
-              'src/base/ftcid.c',
-              'src/base/ftdbgmem.c',
-              'src/base/ftdebug.c',
-              'src/base/ftfntfmt.c',
-              'src/base/ftfstype.c',
-              'src/base/ftgasp.c',
-              'src/base/ftgloadr.c',
-              'src/base/ftglyph.c',
-              'src/base/ftgxval.c',
-              'src/base/ftinit.c',
-              'src/base/ftlcdfil.c',
-              'src/base/ftmm.c',
-              'src/base/ftobjs.c',
-              'src/base/ftotval.c',
-              'src/base/ftoutln.c',
-              'src/base/ftpatent.c',
-              'src/base/ftpfr.c',
-              'src/base/ftrfork.c',
-              'src/base/ftsnames.c',
-              'src/base/ftstream.c',
-              'src/base/ftstroke.c',
-              'src/base/ftsynth.c',
-              'src/base/ftsystem.c',
-              'src/base/fttrigon.c',
-              'src/base/fttype1.c',
-              'src/base/ftutil.c',
-              'src/base/ftwinfnt.c',
-              'src/bdf/bdf.c',
-              'src/bzip2/ftbzip2.c',
-              'src/cache/ftcache.c',
-              'src/cff/cff.c',
-              'src/cid/type1cid.c',
-              'src/gzip/ftgzip.c',
-              'src/lzw/ftlzw.c',
-              'src/pcf/pcf.c',
-              'src/pfr/pfr.c',
-              'src/psaux/psaux.c',
-              'src/pshinter/pshinter.c',
-              'src/psnames/psmodule.c',
-              'src/raster/raster.c',
-              'src/sfnt/sfnt.c',
-              'src/smooth/smooth.c',
-              'src/truetype/truetype.c',
-              'src/type1/type1.c',
-              'src/type42/type42.c',
-              'src/winfonts/winfnt.c']
-      commands = []
-      o_s = []
-      for src in srcs:
-        o = os.path.join(ports.get_build_dir(), 'freetype', src + '.o')
-        shared.safe_ensure_dirs(os.path.dirname(o))
-        commands.append([shared.PYTHON, shared.EMCC, os.path.join(dest_path, src), '-DFT2_BUILD_LIBRARY', '-O2', '-o', o, '-I' + dest_path + '/include',
-                         '-I' + dest_path + '/truetype', '-I' + dest_path + '/sfnt', '-I' + dest_path + '/autofit', '-I' + dest_path + '/smooth', 
-                         '-I' + dest_path + '/raster', '-I' + dest_path + '/psaux', '-I' + dest_path + '/psnames', '-I' + dest_path + '/truetype', 
-                         '-w',])
-        o_s.append(o)
-
-      ports.run_commands(commands)
-      final = os.path.join(ports.get_build_dir(), 'freetype', 'libfreetype.a')
-      shared.try_delete(final)
-      Popen([shared.LLVM_AR, 'rc', final] + o_s).communicate()
-      assert os.path.exists(final)
-      return final
+      shared.Building.configure(['cmake', '-H' + source_path, '-B' + dest_path,
+        '-DCMAKE_BUILD_TYPE=Release', '-DCMAKE_INSTALL_PREFIX:PATH=' + dest_path])
+      shared.Building.make(['make', '-C' + dest_path])
```

Comment:
I think this line  builds all the sources? It would be useful to pass it a `-jNUM` param to use multiple cores. I opened #6420 now which provides a method to get the proper number, using `shared.Building.get_num_cores()`

Reply:
Yes. Oh, thats great, subscribing to the PR.